qiskit_nature/circuit/library/ansaetze/excitation_builder.py

```diff
@@ -0,0 +1,115 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+TODO.
+"""
+
+from typing import List, Tuple
+
+import itertools
+import logging
+
+from qiskit_nature.operators.second_quantization import FermionicOp, SecondQuantizedOp
+
+logger = logging.getLogger(__name__)
+
+
+class ExcitationBuilder:
+    """A factory class to construct excitation operators."""
+
+    @staticmethod
+    def build_excitation_ops(num_excitations: int,
```

Comment:
Should the method name include `fermionic` in it, or `molecular`. Maybe could build ops for lattices too etc.

Reply:
I was going to talk to Pauline about the UVCC Ansatz and see how much of this method we can potentially reuse. If we can reuse none of it, I will make it a fermionic version.
If, however, we can reuse the excitation generation logic, I will extract the conversion to the operator in order to not imply the fermionic property.